model/model_test.go

```diff
@@ -122,3 +123,32 @@ func TestModel_AddDef(t *testing.T) {
 		t.Errorf("empty assertion value should not be added")
 	}
 }
+
+func TestModelToText(t *testing.T) {
+	m := NewModel()
+	data := map[string]string{
+		"r": "sub, obj, act",
+		"p": "sub, obj, act",
+		"e": "some(where (p.eft == allow))",
+		"m": "r.sub == p.sub && r.obj == p.obj && is_rw(r.act, p.act)",
+	}
+	expected := map[string]string{
+		"r": "sub, obj, act",
+		"p": "sub, obj, act",
+		"e": "some(where (p_eft == allow))",
+		"m": "r_sub == p_sub && r_obj == p_obj && is_rw(r_act, p_act)",
```

Comment:
please add test `m = r_sub == p_sub && r_obj == p_obj && r_fuc(r_act, p_act) && testr_func(r_act, p_act)`.

Reply:
fixed

Reply:
@closetool don't remove the `r_func()` and `testr_func()`, which used to test `rPattern := regexp.MustCompile(`r_`)`, and you should also test the `regexp.MustCompile(`p_`)`.

Reply:
@closetool Please update the test.

Reply:
@nodece There's still some problems, we cann't figure out if it's a policy token or a user defined function for pattern has `p_` and `r_`, so `testr_func` will lead to error.

Reply:
@closetool I found this problem, so I asked to use this master for test.

Reply:
@nodece fixed